src/libponyc/codegen/genexe.cc

```diff
@@ -281,6 +286,106 @@ static const char* env_cc_or_pony_compiler(bool* out_fallback_linker)
 }
 #endif
 
+/*
+ * search_path (called from search_paths)
+ *  - A single tuple (path, depth) to search
+ *  - The target filename to find
+ *  - Whether to include the filename or not
+ *
+ * There are two types of searches that we do.  One to find the paths
+ * so we can add them via -L (libpthread, libm, librt, libatomic etc),
+ * the other for identifying the paths to specific object files such
+ * as crtn.o et al).
+ *
+ * I think this should be refactored into two separate functions as
+ * we should probably deduplicate the -L paths before push_backing them
+ * onto args.
+ */
+std::optional<std::string> search_path(std::tuple<fs::path, int> search_path,
```

Comment:
we don't use optional elsewhere. you aren't using it all the code here. i think we should  not use optional.